### ISSUE TITLE
Re-design nav bar

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#3651]
### Why/User Benefit/User Problem
Results of the latest perceived performance user study included mixed feedback on the dynamic bottom nav bar. We switched to a static nav bar until we investigate and re-design behaviour. This meta will track the re-design and all related issues.

Issues related to implementing dynamic bottom nav bar:
- Content partially covered by the quick action bar #2563 

### Acceptance Criteria
-UX completed
-User stories completed
-QA completed

### Reference Issues
https://github.com/mozilla-mobile/fenix/issues/2115
move address bar to bottom of homepage: https://github.com/mozilla-mobile/fenix/issues/1936
set position of search bar: https://github.com/mozilla-mobile/fenix/issues/466
bar moves around depending on context: https://github.com/mozilla-mobile/fenix/issues/562

**Comment by mheubusch** [comment]
@apbitner Assigning to you to consider while you look at nav.

**Comment by apbitner** [comment]
@vesta0 can you define what you mean by 'dynamic nav bar' and 'static nav bar' in this instance?

**Comment by lime124** [comment]
@vesta0 is this labeled waiting b/c we need ux? if so, can we remove the waiting and ux feedback labels as Adam is already working on this. thanks!

**Comment by liuche** [comment]
For my own notes:
GV couldn't handle bottom overlapping nav bar w/ autohiding (and the resizing transition), so some changes were needed.

**Comment by liuche** [comment]
This could also be a good candidate for some UX/Eng collab, like suggested in #419 .

**Comment by lime124** [comment]
removed ux label -it's on our radar already as it's part of the Q3 plan

**Comment by apbitner** [comment]
UX explorations on InVision: https://mozilla.invisionapp.com/share/W8SZ87HSN2G

The top desired options involve auto show/hide of toolbars, so UX work is currently on hold while feasibility and LOE of options is determined with GV team.

**Comment by Madis0** [comment]
I like the options C and D. Make it either top or bottom, not both!

**Comment by pwd-github** [comment]
Not sure if my vote counts, but I'd vote for option D.

**Comment by liuche** [comment]
@apbitner could you give an update here? Are these proposals feasible with the GV options, or should we go with some fallbacks, or altogether deprioritize this until GV is ready?

**Comment by apbitner** [comment]
@liuche We chatted with @vesta0 yesterday and GV said it would take months to implement the auto show/hide we're hoping for. Vesta is waiting on detailed documentation from GV including possible alternative options. UX has alternative options of our own but they would require getting rid of the QAB entirely.

**Comment by robsmith11** [comment]
Why not just get rid of the QAB if that helps implement a better nav bar?  The QAB functionality is redundant as it is.

**Comment by cadeyrn** [comment]
> The QAB functionality is redundant as it is.

This is not true at all. Or how do you add bookmarks, open the reading mode or use the "open with app" feature? With the current implementation there is no other way to access these feature.

**Comment by andreicristianpetcu** [comment]
QAB is a great place to put WebExtension actions

**Comment by liuche** [comment]
@cpeterso what is the GV or Gecko bug related to this? I know this bug has been around forever but for some reason I can't find it 😅

**Comment by cpeterso** [comment]
> what is the GV or Gecko bug related to this? I know this bug has been around forever but for some reason I can't find it 😅

You can't find the bug because there isn't one. 😄

We have a couple GV/Gecko bugs for dynamic nav bar issues (below), but we didn't open a bug for the big compositor rewrite because we decided against that approach. The rewrite might take about six months so we decided address individual Gecko bugs as Fenix finds them after re-enabling the dynamic nav bar.

Some known issues:

* Fix `setVerticalClipping` API (to be fixed in September): https://bugzilla.mozilla.org/show_bug.cgi?id=1552608
* Fix `setVerticalClipping` API for WebRender: https://bugzilla.mozilla.org/show_bug.cgi?id=1583380
* Element hit boxes are too low: https://bugzilla.mozilla.org/show_bug.cgi?id=1555446
* Bottom fixed margin layers bug (fixed in GV 67): https://bugzilla.mozilla.org/show_bug.cgi?id=1546139
* Bottom fixed position bug (fixed in GV 68): https://bugzilla.mozilla.org/show_bug.cgi?id=1516048

**Comment by apbitner** [comment]
QAB will be removed for now, and revisited once we have auto show/hide capabilities. This will make implementing a top nav easier, though we will have to make updates to the nav bar menu.

**Comment by lime124** [comment]
thanks @apbitner!  

this is what i think is left to complete before passing off to eng. adam, please edit this list with whatever i missed!
- setting for top vs bottom nav
- doing this ticket to add in missing QAB items: #4281 
-  final specs and hand off meeting

 I know nicole is doing some stuff in the menu, can you coordinate with her so we are getting all the changes we need in one spec/hand off? thanks!

**Comment by lime124** [comment]
@cpeterso i'm not totally following your comment and i just want to make sure we all have the same understanding :)

my current understanding is that GV will do the work needed to have the auto show/hide capabilities in Q1. is that your understanding too? thanks!

**Comment by cpeterso** [comment]
> my current understanding is that GV will do the work needed to have the auto show/hide capabilities in Q1. is that your understanding too? thanks!

We have 2-3 known Gecko/GV bugs that block Fenix auto showing/hiding the dynamic nav bar. GV will be working on those bugs in early Q4. Fenix should be able to re-enable the dynamic nav bar in Fenix Nightly for some pre-release user testing in Q4.

**Comment by lime124** [comment]
Cool thanks!

**Comment by vesta0** [comment]
As discussed during planning week, the actionable next step here is to provide users with the option to move nav bar to the top. Created #5694 and closing this one. 

Turning on the dynamic bottom nav bar down the road will be addressed here: #4383

Removing the QAB will be addressed here: #5588 

@lime124 and @apbitner let me know if I have missed anything.

**Comment by kmkaplan** [comment]
The reference issue  "bar moves around depending on context" #562 does not seem to be addressed. Did I miss something?
